cppwinrt/code_writers.h

```diff
@@ -1310,15 +1310,15 @@ namespace cppwinrt
             if constexpr (std::is_base_of_v<Windows::Foundation::IUnknown, V>)
             {
                 V result{ nullptr };
-                WINRT_IMPL_SHIM(Windows::Foundation::Collections::IMapView<K, V>)->Lookup(get_abi(key), put_abi(result));
+                impl::check_hresult_allow_bounds(WINRT_IMPL_SHIM(Windows::Foundation::Collections::IMapView<K, V>)->Lookup(get_abi(key), put_abi(result)));
```

Comment:
This throws and yet `TryLookup` has always been `noexcept`.

Reply:
Good catch. I need to remove the noexcept from here and TryRemove. The point is that there are three outcomes:

1. Item is definitely present (and was looked up / removed)
2. Item is definitely missing (and was not looked up / not removed)
3. Operation was unsuccessful

Right now, cases 2 and 3 are conflated, which means that you have bugs where the remote server is unavailable (maybe it is hung or broken into the debugger and you got `RPC_E_TIMEOUT`), and you do a TryLookup, see that it's not there, and assume "Well, then that means it was never added, so I don't need to clean up my client tracking data." But it actually means "I couldn't perform the desired operation at all. The thing you added might or might not be there, we can't tell."